pkgs/tools/typesetting/tex/nix/default.nix

```diff
@@ -43,11 +44,11 @@ rec {
   # \input{}), images (e.g. \includegraphics{}), bibliographies, and
   # so on.
   findLaTeXIncludes =
-    { rootFile
+    { rootFile, searchPath
     }:
 
     builtins.genericClosure {
-      startSet = [{key = rootFile;}];
+      startSet = [{key = toString rootFile;}];
```

Comment:
You said, not having this breaks things. Do you understand why?

Reply:
This was the first time I was made aware of string contexts in Nix: I started getting errors like `cannot compare a path with a string with context`.

From what I have gathered, the problem is that in the implementation of `builtins.genericClosure` the values of `key` are compared with each other, and Nix, being a strictly typed dynamic language, sees strings with a context and ordinary strings(paths) as distinct types, and throws an error. 

I have experimentally found that explicitly casting `rootFile` to a string (calling `toString`) seems to avoid the error (at least on the inputs I have). However, I have not been able to track down exactly where and why the string context appears. Because of this, I am not entirely confident that this code will work with all seemingly valid input. 

In my estimation, casting all values of `key` to strings explicitly (calling `toString`) should work in all cases since `key` will always be the same type. This is the reason I expressed my doubts in the comment: https://github.com/NixOS/nixpkgs/pull/119883#discussion_r616079867.